CHANGELOG.md

```diff
@@ -42,6 +42,16 @@
 - Changed the dump feature of `cpu-template-helper` tool not to enumerate program
   counter (PC) on ARM because it is determined by the given kernel image and
   it is useless in the custom CPU template context.
+- We use a new version of `userfaultfd-rs`, which changes the way that a
+  userfault file descriptor (UFFD) is created when running on (host) Linux
+  kernels >=6.1. In these kernels the library first tries to create the UFFD
+  through an `ioctl()` to the `/dev/userfaultfd` device. If the device is not
+  present, the library falls back to calling the `userfaultfd` system call.
+  Access to `/dev/userfaultfd` is managed through normal filesystem
+  permissions. That means that, when using the UFFD functionality on Linux
+  kernel 6.1, users need to ensure that the Firecracker process has sufficient
+  permissions to access `/dev/userfaultfd`. This change does not affect users
+  that use UFFD on host kernels 4.14 and 5.10.
```

Comment:
would it be possible to use kernel version ranges instead? Customers may use anything in between 4.14 and 6.1 too.

Reply:
I wanted to explicitly use the kernel versions we support, so that we are explicit about that. Do you think we should add documentation for non-supported versions?

Reply:
> Linux kernels >=6.1

This kind of already goes beyond that. I think at least we can be consistent with either saying what happens only on 4.14, 5.10, 6.1 or covering a range. This is more of a nit though.

Reply:
Actually, the more I read this, the more I think it should better go to the doc folder, and the changelog summary could be more concise.

Reply:
> > Linux kernels >=6.1
> 
> This kind of already goes beyond that. I think at least we can be consistent with either saying what happens only on 4.14, 5.10, 6.1 or covering a range. This is more of a nit though.

>= 6.1 refers to what Linux does. When I refer to 4.14, 5.10 & 6.1 later, I refer to what Firecracker supports. I'd like to distinguish between the two. Mainly, because I'd like to avoid having to debug cases of users on the non-supported ranges.

Reply:
> Actually, the more I read this, the more I think it should better go to the doc folder, and the changelog summary could be more concise.

I can try to make this more concise, but I'd like to have _some_ info about the fact that users need to take care their `/dev/userfaultfd` permissions in the CHANGELOG. WDYT?

Reply:
FYI, I addressed this according to a suggestion from Patrick: https://github.com/firecracker-microvm/firecracker/pull/4086#discussion_r1334560717